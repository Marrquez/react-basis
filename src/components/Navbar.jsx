function Navbar() {
    return <h1>Hello world from Navbar! :D</h1>
}

export default Navbar;