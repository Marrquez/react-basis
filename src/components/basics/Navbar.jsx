import React, { useState } from "react";

function Navbar() {
    const [heading, setHeading] = useState("Magnificent Monkeys");

    const clickHandler = () => {
        setHeading("Radical Rhinos");
    };

    return (
        <>
            <h1>Hello world from Navbar</h1>
            <button type="button" onClick={clickHandler}>
                Click Me
            </button>
            <p>{heading}</p>
            <nav>
                <ul>
                    <li>
                        <a href="/">Home</a>
                    </li>
                    <li>
                        <a href="/dashboard">Dashboard</a>
                    </li>
                    <li>
                        <a href="login">Login page</a>
                    </li>
                </ul>
            </nav>
        </>
    );
}

export default Navbar;