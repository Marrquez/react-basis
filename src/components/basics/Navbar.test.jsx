import { vi, describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Navbar from "./Navbar";

describe("Navbar component", () => {
  it("renders correct heading", () => {
    render(<Navbar />);
    expect(screen.getByRole("heading").textContent).toMatch(/Hello world from Navbar/i);
  });

  it("renders magnificent monkeys", () => {
    // since screen does not have the container property, we'll destructure render to obtain a container for this test
    const { container } = render(<Navbar />);
    expect(container).toMatchSnapshot();
  });

  it("renders radical rhinos after button click", async () => {
    const user = userEvent.setup();

    render(<Navbar />);
    const button = screen.getByRole("button", { name: "Click Me" });

    await user.click(button);

    expect(screen.getByRole("paragraph").textContent).toMatch(/Radical Rhinos/i);
  });

//   it("should call the onClick function when clicked", async () => {
//     const onClick = vi.fn();
//     const user = userEvent.setup()
    
//     render(<Navbar />);
//     const button = screen.getByRole("button", { name: "Click Me" });

//     await user.click(button);

//     expect(onClick).toHaveBeenCalled();
//   });

  it("should not call the onClick function when it isn't clicked", async () => {
    const onClick = vi.fn();
    render(<Navbar />);

    expect(onClick).not.toHaveBeenCalled();
  });
});