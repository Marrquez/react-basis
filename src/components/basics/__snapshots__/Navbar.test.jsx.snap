// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar component > renders magnificent monkeys 1`] = `
<div>
  <h1>
    Hello world from Navbar
  </h1>
  <button
    type="button"
  >
    Click Me
  </button>
  <p>
    Magnificent Monkeys
  </p>
</div>
`;
